$NetBSD: patch-ae,v 1.6 2026/10/25 15:27:12 taca Exp $

* Use watcher specified by -m option if available.
* Pass correct type to ctype(3) function.

--- report.c.orig	2000-09-30 23:41:10.000000000 +0000
+++ report.c
@@ -70,6 +70,8 @@ struct rtentry;
 
 #define PLURAL(n) ((n) == 1 || (n) == -1 ? "" : "s")
 
+char *Watcher;
+
 static int cdepth;	/* number of outstanding children */
 
 static char *fmtdate(time_t);
@@ -240,7 +242,7 @@ report(register char *title, register u_
 	register FILE *f;
 	char tempfile[64], cpu[64], os[64];
 	char *fmt = "%20s: %s\n";
-	char *watcher = WATCHER;
+	char *watcher = Watcher ? Watcher : WATCHER;
 	char *watchee = WATCHEE;
 	char *sendmail = PATH_SENDMAIL;
 	char *unknown = "<unknown>";
@@ -303,7 +305,7 @@ report(register char *title, register u_
 	(void)fprintf(f, "From: %s\n", watchee);
 	(void)fprintf(f, "To: %s\n", watcher);
 	hn = gethname(a);
-	if (!isdigit(*hn))
+	if (!isdigit((unsigned char)*hn))
 		(void)fprintf(f, "Subject: %s (%s)\n", title, hn);
 	else {
 		(void)fprintf(f, "Subject: %s\n", title);
